tests/autologging/test_autologging_utils.py

```diff
@@ -642,3 +645,143 @@ def test_autologging_event_logger_default_impl_warns_for_log_autolog_called_with
+        ("pytorch", {"torch": "1.3.9", "pytorch_lightning": "1.0.5"}, False),
+        ("pytorch", {"torch": "1.4.0", "pytorch_lightning": "1.0.4"}, False),
+    ],
+)
+@mock.patch(
+    "mlflow.utils.autologging_utils._module_version_info_dict", _module_version_info_dict_patch
+)
+def test_is_autologging_integration_supported(flavor, module_version_dict, expected_result):
+    with ExitStack() as stack:
+        for module_name in module_version_dict:
+            stack.enter_context(
+                mock.patch(module_name + ".__version__", module_version_dict[module_name])
+            )
+        assert expected_result == is_autologging_integration_supported(flavor)
+
+
+@mock.patch(
+    "mlflow.utils.autologging_utils._module_version_info_dict", _module_version_info_dict_patch
+)
+def test_disable_for_unsupported_versions_warning_sklearn_integration():
+    log_warn_fn_name = "mlflow.utils.autologging_utils._log_warning_for_unsupported_integration"
+
+    with mock.patch("sklearn.__version__", "0.20.3"):
+        AUTOLOGGING_INTEGRATIONS.clear()
+        with mock.patch(log_warn_fn_name) as log_warn_fn:
+            mlflow.sklearn.autolog(disable_for_unsupported_versions=True)
+            with pytest.raises(AssertionError):
+                log_warn_fn.assert_called_once_with("sklearn")
+        AUTOLOGGING_INTEGRATIONS.clear()
+        with mock.patch(log_warn_fn_name) as log_warn_fn:
+            mlflow.sklearn.autolog(disable_for_unsupported_versions=False)
+            with pytest.raises(AssertionError):
+                log_warn_fn.assert_called_once_with("sklearn")
+
+        AUTOLOGGING_INTEGRATIONS.clear()
+        with mock.patch(log_warn_fn_name) as log_warn_fn:
+            mlflow.autolog(disable_for_unsupported_versions=True)
+            with pytest.raises(AssertionError):
+                log_warn_fn.assert_any_call("sklearn")
+        AUTOLOGGING_INTEGRATIONS.clear()
+        with mock.patch(log_warn_fn_name) as log_warn_fn:
+            mlflow.autolog(disable_for_unsupported_versions=False)
+            with pytest.raises(AssertionError):
+                log_warn_fn.assert_any_call("sklearn")
+
+    with mock.patch("sklearn.__version__", "0.20.2"):
+        AUTOLOGGING_INTEGRATIONS.clear()
+        with mock.patch(log_warn_fn_name) as log_warn_fn:
+            mlflow.sklearn.autolog(disable_for_unsupported_versions=True)
+            with pytest.raises(AssertionError):
+                log_warn_fn.assert_called_once_with("sklearn")
+        with mock.patch(log_warn_fn_name) as log_warn_fn:
+            mlflow.sklearn.autolog(disable_for_unsupported_versions=False)
+            log_warn_fn.assert_called_once_with("sklearn")
+
+        AUTOLOGGING_INTEGRATIONS.clear()
+        with mock.patch(log_warn_fn_name) as log_warn_fn:
+            mlflow.autolog(disable_for_unsupported_versions=True)
+            with pytest.raises(AssertionError):
+                log_warn_fn.assert_any_call("sklearn")
```

Comment:
I think you can use `assert_not_called` here:
https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_not_called

Reply:
No. It is possible being called, with other flavor_name.
because the universal autolog function will hook all flavor libraries.

Reply:
btw what are you asserting here?

Reply:
call_args_list